sonaranalyzer-dotnet/src/SonarAnalyzer.VisualBasic/Metrics/CognitiveComplexityWalker.cs

```diff
@@ -0,0 +1,210 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.VisualBasic;
+using Microsoft.CodeAnalysis.VisualBasic.Syntax;
+using SonarAnalyzer.Helpers.VisualBasic;
+
+namespace SonarAnalyzer.Common.VisualBasic
+{
+    public sealed class CognitiveComplexityWalker : CognitiveComplexityWalkerBase
+    {
+        private MethodStatementSyntax currentMethod;
+        private readonly List<ExpressionSyntax> logicalOperationsToIgnore = new List<ExpressionSyntax>();
```

Comment:
We don't really care about the `ExpressionSyntax` type so you could move this collection to the base type and use `SyntaxNode` instead.